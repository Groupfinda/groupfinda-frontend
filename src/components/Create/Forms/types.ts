import { CustomError } from "../../../hooks";
import { ImagePickerResult } from "expo-image-picker";

export type FormProps = {
  variables: FormVariablesType;
  modifyVariable: (key: string) => (value: any) => void;
  prevPage?: () => void;
  nextPage?: () => void;
};

type ValidateProps = {
  validateFieldLength: (
    key: string
  ) => (setCustomError: (err: CustomError) => void) => () => boolean;
};

export type FormPropsWithValidate = FormProps & ValidateProps;

export type FormVariablesType = {
  title: string;
  description: string;
  dateOfEvent: Date;
  recurringMode: boolean;
  dateLastRegister: Date;
  images: string[];
  privateStatus: boolean;
  groupSize: number;
  category: string[];
  locationOn: boolean;
  address: string;
  postalCode: string;
};
