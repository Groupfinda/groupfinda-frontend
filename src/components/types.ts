import { Input, Select } from "@ui-kitten/components";

export type ReferencesType = {
  [key: string]: Input | Select | null;
};

export type QuestionType = {
  order: number,
  content: string,
  value: number
};

export type BasicEventType = {
  id: string,
  title: string,
  dateOfEvent: number
}