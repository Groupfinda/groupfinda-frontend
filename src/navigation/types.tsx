import { StackScreenProps } from "@react-navigation/stack";

export type RootStackParamList = {
  LogIn: undefined;
  Main: undefined;
  SignUp: undefined;
  ForgetPassword: undefined;
  ProfileSettings: undefined;
  ChangePassword: undefined;
  Questions: undefined;
  CreateEvent: undefined;
  JoinEvent: undefined;
};

export type SignUpScreenNavigationProp = StackScreenProps<
  RootStackParamList,
  "SignUp"
>;

export type LogInScreenNavigationProp = StackScreenProps<
  RootStackParamList,
  "LogIn"
>;

export type ForgetPasswordScreenNavigationProp = StackScreenProps<
  RootStackParamList,
  "ForgetPassword"
>;

export type ProfileSettingsScreenNavigationProp = StackScreenProps<
  RootStackParamList,
  "ProfileSettings"
>;

export type QuestionsScreenNavigationProp = StackScreenProps<
  RootStackParamList,
  "Questions"
>;

export type CreateEventScreenNavigationProp = StackScreenProps<
  RootStackParamList,
  "CreateEvent"
>;

export type JoinEventScreenNavigationProp = StackScreenProps<
  RootStackParamList,
  "JoinEvent"
>;
