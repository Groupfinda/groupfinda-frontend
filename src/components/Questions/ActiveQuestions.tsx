import React from "react";
import { ScrollView, View } from "react-native";
import { QuestionCard } from './extra/question-card.component'
import { Layout, StyleService, useStyleSheet, Button, Text, Icon, IconElement } from "@ui-kitten/components";

const questions = [
    {
        order: 15,
        question: "I am an efficient person"
    },
    {
        order: 16,
        question: "I enjoy being in the company of others"
    },
    {
        order: 17,
        question: "I love seeking out new adventures"
    }
]

export default (): React.ReactElement => {

    const styles = useStyleSheet(themedStyle)
    const [ currentIndex, setIndex ] = React.useState(0);
    const [ selectedIndex, setSelectedIndex ] = React.useState(0);

    const handleSubmit = () => {
        // post here, await response then go to next question
        // console.log(selectedIndex)
        // console.log(questions[currentIndex].order)
        if (currentIndex + 1 < questions.length) {
            setIndex(currentIndex + 1)
            setSelectedIndex(0)
        }
    }

    const skipQuestion = () => {
        setIndex(currentIndex + 1)
        setSelectedIndex(0)
    }

    const previousQuestion = () => {
        setIndex(currentIndex - 1)
        setSelectedIndex(0)
    }

    return (
        <ScrollView style={styles.container}>
            <Layout style={styles.cardContainer}>
               <QuestionCard
                    order={questions[currentIndex].order}
                    question={questions[currentIndex].question}
                    selectedIndex={selectedIndex}
                    setSelectedIndex={setSelectedIndex}/>
                <Button
                    status='success'
                    disabled={selectedIndex<1||selectedIndex>5}
                    style={styles.buttonStyle}
                    onPress={handleSubmit}>
                    Save Answer
                </Button>
            </Layout>
            <Layout style={styles.skipContainer}>
                <Layout level='1'>
                    <Button
                        style={{"width": 100}}
                        accessoryLeft={BackIcon}
                        status='primary'
                        onPress={()=>previousQuestion()}
                        disabled={currentIndex<=0}>Previous</Button>
                </Layout>
                <Layout level='2' style={styles.numberContainer}>
                    <Text appearance='alternative'
                        status='control'>{questions[currentIndex].order}/50</Text>
                </Layout>
                <Layout level='3'>
                    <Button
                        style={{"width": 100}}
                        accessoryRight={NextIcon}
                        status='primary'
                        onPress={()=>skipQuestion()}
                        disabled={!(currentIndex + 1 < questions.length)}>
                        Skip
                    </Button>
                </Layout>
            </Layout>
        </ScrollView>
    )
};

const BackIcon = (): IconElement => {
    return (
        <Icon
            width={18}
            height={18}
            fill='white'
            name='arrow-ios-back'/>
    )
}

const NextIcon = (): IconElement => {
    return (
        <Icon
            width={18}
            height={18}
            fill='white'
            name='arrow-ios-forward'/>
    )
}

const themedStyle = StyleService.create({
    container: {
        backgroundColor: "background-basic-color-3"
    },
    cardContainer: {
        padding: 10,
        backgroundColor: "background-basic-color-3"
    },
    buttonStyle: {
        marginTop: 15
    },
    skipContainer: {
        padding: 15,
        backgroundColor: "transparent",
        flexDirection: "row",
        justifyContent: "space-between"
    },
    numberContainer: {
        padding: 10,
        backgroundColor: "color-primary-default",
        borderRadius: 8,
        textAlign: "center"
    }
})