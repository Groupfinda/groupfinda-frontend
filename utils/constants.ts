export const eventCategories = [
  "Outdoors",
  "Adventure",
  "Technology",
  "Health",
  "Sports",
  "Fitness",
  "Learning",
  "Food & Drink",
  "Culture",
  "Music",
  "Social",
  "Games",
  "Arts",
  "Pets",
  "Hobbies",
  "Volunteer",
];
