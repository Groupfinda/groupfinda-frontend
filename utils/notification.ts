import Constants from 'expo-constants'
import * as Permissions from 'expo-permissions'
import { Platform } from "react-native";
import * as Notifications from 'expo-notifications'

export async function registerForPushNotificationsAsync() {
    let token;
    if (Constants.isDevice) {
        const { status: existingStatus } = await Permissions.getAsync(Permissions.NOTIFICATIONS);
        let finalStatus = existingStatus;
        if (existingStatus !== 'granted') {
            const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
            finalStatus = status;
        }

        if (finalStatus !== 'granted') {
            alert("Notifications allow us to send you updates on group matches and chat messages!")
            return
        }

        token = (await Notifications.getExpoPushTokenAsync()).data
        console.log(token)
    }

    if (Platform.OS === 'android') {
        Notifications.setNotificationChannelAsync('Groupfinda', {
            name: 'Groupfinda',
            importance: Notifications.AndroidImportance.MAX,
            vibrationPattern: [0, 250, 250, 250],
            lightColor: "#FF231F7C"
        })
    }

    return token
}

export type GroupfindaBody = {
    id: string;
    type: string;
}

export type AndroidNotificationData = GroupfindaBody
export type IOSNotificationData = {
    aps: {
        alert: {
            body: string;
            "launch-image": string;
            subtitle: string;
            title: string;
        }
        category: string;
        sound: string;
        "thread-id": string;
    }
    body: GroupfindaBody
}