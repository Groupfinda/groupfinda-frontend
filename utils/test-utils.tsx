import React from "react";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { myTheme } from "../custom-theme";
import { EvaIconsPack } from "@ui-kitten/eva-icons";
import { MockedProvider, MockedResponse } from "@apollo/react-testing";
import { render, RenderAPI, RenderOptions } from "react-native-testing-library";

const AllProviders = (apolloMocks: MockedResponse[]) => ({
  children,
}: {
  children: JSX.Element;
}) => {
  return (
    <>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={{ ...eva.light, ...myTheme }}>
        <MockedProvider mocks={apolloMocks} addTypename={false}>
          {children}
        </MockedProvider>
      </ApplicationProvider>
    </>
  );
};

type CustomRenderType = (
  component: React.ReactElement<any>,
  options?: RenderOptions,
  apolloProps?: MockedResponse[]
) => RenderAPI;

const customRender: CustomRenderType = (
  ui,
  options,
  apolloProps: MockedResponse[] = []
) => render(ui, { wrapper: AllProviders(apolloProps), ...options });

export * from "react-native-testing-library";
export { customRender as render };
