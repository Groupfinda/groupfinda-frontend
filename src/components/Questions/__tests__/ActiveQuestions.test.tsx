import React from "react";
import { render, cleanup, fireEvent, act } from "../../../../utils/test-utils"
import ActiveQuestions from "../ActiveQuestions";
import { QuestionType } from "../../types"

afterEach(cleanup);

let questions: QuestionType[] = []
let completedQuestions: QuestionType[] = []
let setCompletedQuestions: any
for (let i=6; i < 300; i++) {
    questions.push({
        order: i,
        content: "Dummy Question",
        value: 0
    })
}

test("renders UI correctly", () => {
    const tree = render(<ActiveQuestions questions={questions} completedQuestions={completedQuestions} setCompletedQuestions={setCompletedQuestions}/>);
    expect(tree.toJSON()).toMatchSnapshot()
})