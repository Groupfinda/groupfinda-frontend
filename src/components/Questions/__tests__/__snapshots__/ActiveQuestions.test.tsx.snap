// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#EDF1F7",
      }
    }
  >
    <View>
      <View
        appearance="default"
        level="1"
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
            },
            Object {
              "backgroundColor": "#EDF1F7",
              "padding": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#006ECB",
              "borderColor": "#E4E9F2",
              "borderRadius": 4,
              "borderWidth": 1,
              "opacity": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": undefined,
                "height": 0,
              }
            }
          />
          <View
            style={
              Object {
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <View
              appearance="default"
              level="1"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                  },
                  undefined,
                ]
              }
            >
              <Text
                appearance="default"
                category="h6"
                status="control"
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 18,
                      "fontWeight": "800",
                    },
                    Object {
                      "backgroundColor": "#006ECB",
                    },
                  ]
                }
              >
                7
                . 
                Dummy Question
              </Text>
            </View>
          </View>
          <View
            appearance="default"
            style={
              Array [
                Object {
                  "backgroundColor": "#EDF1F7",
                  "height": 1,
                },
                undefined,
              ]
            }
          />
          <View
            appearance="default"
            level="1"
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                Object {
                  "backgroundColor": "#CBF5FE",
                },
              ]
            }
          >
            <View
              appearance="default"
              style={
                Array [
                  Object {
                    "backgroundColor": "#EDF1F7",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "opacity": 1,
                  "paddingHorizontal": 8,
                  "paddingVertical": 12,
                }
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={20}
                  bbWidth={20}
                  fill="white"
                  focusable={false}
                  height={20}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={20}
                >
                  <RNSVGGroup
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Object {
                      "color": "#0052AA",
                    },
                  ]
                }
              >
                Accurate
              </Text>
            </View>
            <View
              appearance="default"
              style={
                Array [
                  Object {
                    "backgroundColor": "#EDF1F7",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "opacity": 1,
                  "paddingHorizontal": 8,
                  "paddingVertical": 12,
                }
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={20}
                  bbWidth={20}
                  fill="white"
                  focusable={false}
                  height={20}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={20}
                >
                  <RNSVGGroup
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Object {
                      "color": "#0052AA",
                    },
                  ]
                }
              >
                Slightly Accurate
              </Text>
            </View>
            <View
              appearance="default"
              style={
                Array [
                  Object {
                    "backgroundColor": "#EDF1F7",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "opacity": 1,
                  "paddingHorizontal": 8,
                  "paddingVertical": 12,
                }
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={20}
                  bbWidth={20}
                  fill="white"
                  focusable={false}
                  height={20}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={20}
                >
                  <RNSVGGroup
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Object {
                      "color": "#0052AA",
                    },
                  ]
                }
              >
                Neutral
              </Text>
            </View>
            <View
              appearance="default"
              style={
                Array [
                  Object {
                    "backgroundColor": "#EDF1F7",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "opacity": 1,
                  "paddingHorizontal": 8,
                  "paddingVertical": 12,
                }
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={20}
                  bbWidth={20}
                  fill="white"
                  focusable={false}
                  height={20}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={20}
                >
                  <RNSVGGroup
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Object {
                      "color": "#0052AA",
                    },
                  ]
                }
              >
                Slightly Inaccurate
              </Text>
            </View>
            <View
              appearance="default"
              style={
                Array [
                  Object {
                    "backgroundColor": "#EDF1F7",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flexDirection": "row",
                  "opacity": 1,
                  "paddingHorizontal": 8,
                  "paddingVertical": 12,
                }
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={20}
                  bbWidth={20}
                  fill="white"
                  focusable={false}
                  height={20}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={20}
                >
                  <RNSVGGroup
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Object {
                      "color": "#0052AA",
                    },
                  ]
                }
              >
                Inaccurate
              </Text>
            </View>
            <View
              appearance="default"
              style={
                Array [
                  Object {
                    "backgroundColor": "#EDF1F7",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(143, 155, 179, 0.24)",
              "borderColor": "rgba(143, 155, 179, 0.24)",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 15,
              "minHeight": 40,
              "minWidth": 40,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 12,
            }
          }
        >
          <View />
          <Text
            appearance="default"
            category="p1"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                Object {
                  "color": "rgba(143, 155, 179, 0.48)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "marginHorizontal": 10,
                },
              ]
            }
          >
            Save Answer
          </Text>
        </View>
      </View>
      <View
        appearance="default"
        level="1"
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
            },
            Object {
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 15,
            },
          ]
        }
      >
        <View
          appearance="default"
          level="1"
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(143, 155, 179, 0.24)",
                "borderColor": "rgba(143, 155, 179, 0.24)",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 40,
                "minWidth": 40,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
                "width": 100,
              }
            }
          >
            <View
              style={
                Object {
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={18}
                bbWidth={18}
                fill="white"
                focusable={false}
                height={18}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 18,
                      "width": 18,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={18}
              >
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGGroup>
                    <RNSVGGroup>
                      <RNSVGRect
                        height="24"
                        matrix={
                          Array [
                            6.123233995736766e-17,
                            1,
                            -1,
                            6.123233995736766e-17,
                            24,
                            0,
                          ]
                        }
                        opacity={0}
                        width="24"
                        x={0}
                        y={0}
                      />
                      <RNSVGPath
                        d="M13.83 19a1 1 0 0 1-.78-.37l-4.83-6a1 1 0 0 1 0-1.27l5-6a1 1 0 0 1 1.54 1.28L10.29 12l4.32 5.36a1 1 0 0 1-.78 1.64z"
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              appearance="default"
              category="p1"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "rgba(143, 155, 179, 0.48)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Previous
            </Text>
          </View>
        </View>
        <View
          appearance="default"
          level="2"
          style={
            Array [
              Object {
                "backgroundColor": "#F7F9FC",
              },
              Object {
                "backgroundColor": "#008EED",
                "borderRadius": 8,
                "padding": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          <Text
            appearance="alternative"
            category="p1"
            status="control"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                undefined,
              ]
            }
          >
            1
            /
            294
          </Text>
        </View>
        <View
          appearance="default"
          level="3"
          style={
            Array [
              Object {
                "backgroundColor": "#EDF1F7",
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#008EED",
                "borderColor": "#008EED",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 40,
                "minWidth": 40,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
                "width": 100,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Skip
            </Text>
            <View
              style={
                Object {
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={18}
                bbWidth={18}
                fill="white"
                focusable={false}
                height={18}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 18,
                      "width": 18,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={18}
              >
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGGroup>
                    <RNSVGGroup>
                      <RNSVGRect
                        height="24"
                        matrix={
                          Array [
                            6.123233995736766e-17,
                            -1,
                            1,
                            6.123233995736766e-17,
                            0,
                            24,
                          ]
                        }
                        opacity={0}
                        width="24"
                        x={0}
                        y={0}
                      />
                      <RNSVGPath
                        d="M10 19a1 1 0 0 1-.64-.23 1 1 0 0 1-.13-1.41L13.71 12 9.39 6.63a1 1 0 0 1 .15-1.41 1 1 0 0 1 1.46.15l4.83 6a1 1 0 0 1 0 1.27l-5 6A1 1 0 0 1 10 19z"
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
